pkg/repo/index.go

```diff
@@ -324,23 +324,30 @@ func IndexDirectory(dir, baseURL string) (*IndexFile, error) {
 //
 // The source parameter is only used for logging.
 // This will fail if API Version is not set (ErrNoAPIVersion) or if the unmarshal fails.
-func loadIndex(data []byte, source string) (*IndexFile, error) {
+func loadIndex(data []byte, source string, consolidateWarnings bool) (*IndexFile, error) {
 	i := &IndexFile{}
 	if err := yaml.UnmarshalStrict(data, i); err != nil {
 		return i, err
 	}
 
+	invalidCharts := 0
 	for name, cvs := range i.Entries {
 		for idx := len(cvs) - 1; idx >= 0; idx-- {
 			if cvs[idx].APIVersion == "" {
 				cvs[idx].APIVersion = chart.APIVersionV1
 			}
 			if err := cvs[idx].Validate(); err != nil {
-				log.Printf("skipping loading invalid entry for chart %q %q from %s: %s", name, cvs[idx].Version, source, err)
```

Comment:
@schollii is there an indicator here of whether these are new? this is just loading everything from `source`...

Reply:
Doesn't look like it. This function is a pure index-reader, the args dont' have the charts that will be used. 

I tried a helm install here on a chart that is listed as skipped: 
```$ helm install junk myrepo/bad-chart-2018.20.3-0956887.tgz 
...
index.go:339: skipping loading invalid entry for chart "bad-chart" "2018.20.3-0956887"
 from /home/user/.cache/helm/repository/myrepo-index.yaml: 
validation: chart.metadata.version "2018.20.3-0956887" is invalid
Error: failed to download "myrepo/bad-chart-2018.20.3-0956887.tgz" 
(hint: running `helm repo update` may help)
Done
```
In the "failed to download" error message, the hint is incorrect and misleading. If there is a way to return a list of skipped charts from that index function, and make it available to where that download error is shown, then the error could be something like `hint: chart version is invalid, rerun command with --warn-invalid-charts-in-index for details`. 

The flag `consolidateWarnings` is a little vague, can it be renamed to `warnInvalidChartsInIndex`?

Reply:
> The flag `consolidateWarnings` is a little vague, can it be renamed to `warnInvalidChartsInIndex`?

agreed! was waiting for a suggestion of a better name 😀

Reply:
> I tried a helm install here on a chart that is listed as skipped:
> 
> ```
> $ helm install junk myrepo/bad-chart-2018.20.3-0956887.tgz 
> ...
> index.go:339: skipping loading invalid entry for chart "bad-chart" "2018.20.3-0956887"
>  from /home/user/.cache/helm/repository/myrepo-index.yaml: 
> validation: chart.metadata.version "2018.20.3-0956887" is invalid
> Error: failed to download "myrepo/bad-chart-2018.20.3-0956887.tgz" 
> (hint: running `helm repo update` may help)
> Done
> ```
> 
> In the "failed to download" error message, the hint is incorrect and misleading. 

Splitting hairs a bit, but it's not necessarily incorrect/misleading in general for this command to say that `helm repo update` ***may*** help. Just not correct in this case. 

The error you posted does include:

```
validation: chart.metadata.version "2018.20.3-0956887" is invalid
```

So it seems pretty clear to me.

Reply:
I did not say that the hint is always wrong, I said it is wrong in this situation where the chart is one of the skipped ones (because updating the repo will not solve the issue). 

Does that "validation" line get printed if your warning flag is off? The point of @bacongobbler was that if hiding the warning that validation error, then the user has no way of knowing what's happened. If that validation error does get printed even with warning off, then that should address @bacongobbler concern and the only change left is a good name for that flag.

Reply:
oh, I misunderstood! I think you're right that

```
validation: chart.metadata.version "2018.20.3-0956887" is invalid
```

was printed as part of the index update and in my case (and others) it would be accompanied by thousands of similar validation errors. those would indeed get consolidated into

```
skipped loading 2964 invalid chart entries from www.helm.repo
```

which is unfortunately also what you'd see if you followed the suggestion of running `helm repo update` (without a flag to enable all validation errors)

perhaps the best (also easiest) way to solve this is for the consolidated message to be something like

```
skipped loading 2964 invalid chart entries from www.helm.repo, run `helm repo update --show-all-warnings` to see each invalid entry
```

Reply:
Yeah that sounds good, if the user sees that summary message and they see the "not downloaded" message right after, they have the info they need I think to dig deeper. 

Are you going to rename the flag to something like `--warn-invalid-charts-in-index`?

Reply:
> Are you going to rename the flag to something like `--warn-invalid-charts-in-index`?

To reiterate, the current default behavior is that (while indexing) the following message is printed for *each* chart/version that fails validation

```
validation: chart.metadata.version "2018.20.3-0956887" is invalid
```

The new default behavior will be to instead show

```
skipped loading 2964 invalid chart entries...
```

And the flag is meant to expand the default summary message into the full set of validation failure messages.

I'm not sure what to call these: "chart validation errors"?

I'm considering something like `--enumerate-validation-errors`.